docs/proposals/upstream-settings.md

```diff
@@ -0,0 +1,210 @@
+# Enhancement Proposal-2467: Upstream Settings Policy
+
+- Issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/2467
+- Status: Implementable
+
+## Summary
+
+This Enhancement Proposal introduces the `UpstreamSettingsPolicy` API that allows Application developers to configure the behavior of the connection between NGINX and their upstream applications. This Policy will attach to a Service that is referenced in an HTTPRoute or GRPCRoute.
+
+## Goals
+
+- Define upstream settings.
+- Define an API for upstream settings.
+
+## Non-Goals
+
+- Provide implementation details for implementing the upstream settings policy.
```

Comment:
Let's mention that we aren't going to tackle upstream settings for TLSRoutes or other layer 4 routes